lib/gemfury/client.rb

```diff
@@ -19,23 +19,11 @@ def account_info
     end
 
     # Uploading a gem file
-    def push_gem(gem_file, options = {})
+    def push_gem(file, options = {})
       ensure_ready!(:authorization)
 
-      # Generate upload link
-      api2 = connection(:api_version => 2)
-      response = api2.post('uploads')
-      checked_response_body(response)
-
-      # Upload to S3
-      upload = response.body['upload']
-      id, s3url = upload['id'], upload['blob']['put']
-      response = s3_put_file(s3url, gem_file)
-      checked_response_body(response)
-
-      # Notify Gemfury that the upload is ready
-      options[:name] ||= File.basename(gem_file.path)
-      response = api2.put("uploads/#{id}", options)
+      push_api = connection(:url => self.pushpoint)
+      response = push_api.post(self.account, options.merge(:file => file))
```

Comment:
Path should be a static `"uploads"`, the account username is passed via the `as` variable

Reply:
Ahh, i now understand the confusion.

In `command/app.rb`, there's this line:

```
opts[:account] = options[:as] if options[:as]
```

So `client.account`, is only set when `--as` option is set in CLI. Otherwise, it's `nil`, and user is identified using the `@user_api_key`, which is either set using the `--api-token` or the saved credentials of the user.

With this setup, i thought using the username as a path will identify the account the package will be pushed to, and the Authorization header to identify the user making the push.

Question now, how will i fill-in the `as=` in HTTP query, if `client.account` will be `nil`?